### PR TITLE
docs(clusters): improve cluster onboarding documentation

### LINKED ISSUE #539: [Docs] - Improve cluster onboarding documentation
> As a ClusterAdmin I want to have documentation so that I know how to onboard a cluster and troubleshoot common issues.

There is already a user guide for [Cluster Onboarding](https://cloudoperators.github.io/greenhouse/docs/user-guides/cluster/onboarding/). The task for this item is to review the documentation, improve wording & examples where necessary. This should aim to make the documentation less technical and guide the enduser through the onboarding process.

Additionally, the guide should cover FAQ's or troubleshooting guidance that can be referred to.

### PR DESCRIPTION
**Pull request by uwe-mayer**
## Description
This aims to better define and outline the permission scopes needed to onboard a `Cluster` to Greenhouse

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Closes #539

### CODE REVIEW

**Review by kengou — CHANGES_REQUESTED** [review]
I'm updating the `greenhousectl cluster onboard ...` just now. Please bear with me there will be just a slight change. Once finished I'm going to update your docs here

File: docs/user-guides/cluster/onboarding.md [hunk 1]
```diff
@@ -6,35 +6,43 @@ description: >
...
 ```
 
 ### Preparation
 
-Download the latest `greenhousectl` binary from [here](https://github.com/cloudoperators/greenhouse/releases). 
+Download the latest `greenhousectl` binary from [here](https://github.com/cloudoperators/greenhouse/releases).
 
-The command line tool requires access to both the Greenhouse **and** your Kubernetes cluster. 
-Hence, have the `kubeconfig` files for both clusters at hand. The `kubeconfig` file for the Greenhouse Kubernetes cluster can be downloaded via the Greenhouse dashboard: _Organization_ > _Clusters_ > _Access greenhouse cluster_. 
+Onboarding a `Cluster` to Greenhouse will require you to authenticate to two different kubernetes clusters via repsective `kubeconfig` files:
```
Comment (IvoGoman) [inline docs/user-guides/cluster/onboarding.md:21]:
```suggestion
Onboarding a `Cluster` to Greenhouse will require you to authenticate to two different Kubernetes clusters via respective `kubeconfig` files:
```

File: docs/user-guides/cluster/onboarding.md [hunk 2]
```diff
@@ -6,35 +6,43 @@ description: >
...
 
-For accessing the **Greenhouse Kubernetes cluster**, the `greenhousectl` will check whether your local kubectl is connected to the Greenhouse Kubernetes cluster. If not connected, 
-either the environment variables `KUBECONFIG` and `KUBECONTEXT` or the *greenhousectl* flags `--kubeconfig` and `--kubecontext`  
-must point to the respective Greenhouse kubeconfig. 
+- `greenhouse`: The cluster your Greenhouse installation is running on. You need `organization-admin` or `cluster-admin` priviledges.
```
Comment (IvoGoman) [inline docs/user-guides/cluster/onboarding.md:23]:
```suggestion
- `greenhouse`: The cluster your Greenhouse installation is running on. You need `organization-admin` or `cluster-admin` privileges.
```

File: docs/user-guides/cluster/onboarding.md [hunk 3]
```diff
@@ -6,35 +6,43 @@ description: >
...
+- `bootstrap`: The cluster you want to onboard. You need `system:masters` priviledges.
```
Comment (IvoGoman) [inline docs/user-guides/cluster/onboarding.md:24]:
```suggestion
- `bootstrap`: The cluster you want to onboard. You need `system:masters` privileges.
```

File: docs/user-guides/cluster/onboarding.md [hunk 4]
```diff
@@ -6,35 +6,43 @@ description: >
...
+
+For consistency we will refer to those two clusters by their names from now on.
+
+You need to have the `kubeconfig` files for both the `greenhouse` and the `bootstrap` cluster at hand. The `kubeconfig` file for the `greenhouse` cluster can be downloaded via the Greenhouse dashboard:
+
+_Organization_ > _Clusters_ > _Access greenhouse cluster_.
```
Comment (IvoGoman) [inline docs/user-guides/cluster/onboarding.md:30]:
```suggestion
_Organization_ > _Clusters_ > _Access Greenhouse cluster_.
```

File: docs/user-guides/cluster/onboarding.md [hunk 5]
```diff
@@ -6,35 +6,43 @@ description: >
...
 
-For accessing your **Kubernetes cluster**, it's mandatory to provide the kubeconfig to the *greenhousectl* using the flag `--bootstrap-kubeconfig` 
 ### Onboard
 
-Since Greenhouse generates URLs which contain the cluster name, we highly recommend to choose a **short** cluster name. 
-In particular for <span style="color:red">Gardener Clusters</span> setting a short name is mandatory, because Gardener has very long cluster names, e.g. `garden-greenhouse--monitoring-external`.
+For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `greenhouse`.
```
Comment (IvoGoman) [inline docs/user-guides/cluster/onboarding.md:34]:
```suggestion
For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default Kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `greenhouse`.
```

File: docs/user-guides/cluster/onboarding.md [hunk 6]
```diff
@@ -6,35 +6,43 @@ description: >
...
+The easiest way for doing so is passing the `--kubeconfig` (and optionally `--kubecontext`) flag to your `greenhousectl` command.
+
+The location of the `kubeconfig` file to the `bootstrap` cluster is passed via the `--bootstrap-kubeconfig` flag.
 
 ```commandline
-greenhousectl cluster bootstrap --bootstrap-kubeconfig <path/to/cluster-kubeconfig-file> --org <greenhouse-organization-name> --cluster-name <name>
+greenhousectl cluster bootstrap --kubeconfig=<path/to/greenhouse-kubeconfig-file> --bootstrap-kubeconfig <path/to/bootstrap-kubeconfig-file> --org <greenhouse-organization-name> --cluster-name <name>
```
Comment (kengou) [inline docs/user-guides/cluster/onboarding.md:40]:
```suggestion
greenhousectl cluster bootstrap --kubeconfig=<path/to/bootstrap-kubeconfig-file> --greenhouse-kubeconfig <path/to/greenhouse-kubeconfig-file> --org <greenhouse-organization-name> --cluster-name <name>
```

File: docs/user-guides/cluster/onboarding.md [hunk 7]
```diff
@@ -6,35 +6,43 @@ description: >
...
 
-For accessing your **Kubernetes cluster**, it's mandatory to provide the kubeconfig to the *greenhousectl* using the flag `--bootstrap-kubeconfig` 
 ### Onboard
 
-Since Greenhouse generates URLs which contain the cluster name, we highly recommend to choose a **short** cluster name. 
-In particular for <span style="color:red">Gardener Clusters</span> setting a short name is mandatory, because Gardener has very long cluster names, e.g. `garden-greenhouse--monitoring-external`.
+For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `greenhouse`.
+The easiest way for doing so is passing the `--kubeconfig` (and optionally `--kubecontext`) flag to your `greenhousectl` command.
+
+The location of the `kubeconfig` file to the `bootstrap` cluster is passed via the `--bootstrap-kubeconfig` flag.
```
Comment (kengou) [inline docs/user-guides/cluster/onboarding.md:37]:
```suggestion
The location of the `kubeconfig` file to the `bootstrap` cluster is passed via the `--kubeconfig` flag or via the `KUBECONFIG` enviorment variable
```

File: docs/user-guides/cluster/onboarding.md (hunk 5 above)
Comment (kengou) [inline docs/user-guides/cluster/onboarding.md:34]:
```suggestion
For accessing the `greenhouse` cluster, the `greenhousectl` will expect you to pass [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) downloaded from Greenhouse dashboard via `--greenhouse-kubeconfig` flag
```

File: docs/user-guides/cluster/onboarding.md [hunk 8]
```diff
@@ -6,35 +6,43 @@ description: >
...
+
+_Organization_ > _Clusters_ > _Access greenhouse cluster_.
 
-For accessing your **Kubernetes cluster**, it's mandatory to provide the kubeconfig to the *greenhousectl* using the flag `--bootstrap-kubeconfig` 
 ### Onboard
 
-Since Greenhouse generates URLs which contain the cluster name, we highly recommend to choose a **short** cluster name. 
-In particular for <span style="color:red">Gardener Clusters</span> setting a short name is mandatory, because Gardener has very long cluster names, e.g. `garden-greenhouse--monitoring-external`.
+For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `greenhouse`.
+The easiest way for doing so is passing the `--kubeconfig` (and optionally `--kubecontext`) flag to your `greenhousectl` command.
```
Comment (kengou) [inline docs/user-guides/cluster/onboarding.md:35]:
```suggestion

```

File: docs/user-guides/cluster/onboarding.md [hunk 9]
```diff
@@ -6,35 +6,42 @@ description: >
...
+You need to have the `kubeconfig` files for both the `greenhouse` and the `bootstrap` cluster at hand. The `kubeconfig` file for the `greenhouse` cluster can be downloaded via the Greenhouse dashboard:
+
+_Organization_ > _Clusters_ > _Access Greenhouse cluster_.
 
-For accessing your **Kubernetes cluster**, it's mandatory to provide the kubeconfig to the *greenhousectl* using the flag `--bootstrap-kubeconfig` 
 ### Onboard
 
-Since Greenhouse generates URLs which contain the cluster name, we highly recommend to choose a **short** cluster name. 
-In particular for <span style="color:red">Gardener Clusters</span> setting a short name is mandatory, because Gardener has very long cluster names, e.g. `garden-greenhouse--monitoring-external`.
+For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default Kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `greenhouse`.
```
Comment (uwe-mayer) [inline docs/user-guides/cluster/onboarding.md:34]:
```suggestion
For accessing the `greenhouse` cluster, the `greenhousectl` will expect your default Kubernetes [`kubeconfig` file](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) and [`context`](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_config/kubectl_config_use-context/) to be set to `bootstrap`.
```

**Review by IvoGoman — APPROVED** [review]

**Review by kengou — APPROVED** [review]

**Pull request merged**